### ISSUE TITLE
Invoking the Context Menu with keyboard combinations does not preserve location information

### ISSUE THREAD

**Issue by robpaveza** [microsoft/monaco-editor#489]
monaco-editor version: 0.8.3
Browser: Edge
OS: Win10
Steps or JS usage snippet reproducing the issue:

1. Add a command handler for a keybinding (for example, Ctrl+F10)
2. Log the arguments that come in for that
Observe: unlike the `onContextMenu` event, there is no target / location information (compare in the "Listening to Mouse Events" sample in the Playground).  Here's one example:

```
    editor.addCommand(monaco.KeyCode.F10, function(e) {
        showEvent('contextmenu keybind - ' + JSON.stringify(arguments));
        for (let i = 0; i < arguments.length; i++) {
            console.log(`dumping ${i}:`);
            for (let key in arguments[i]) {
                console.log(`    ${key}: ${arguments[i][key]}`);
            }
        }
    });
```

This outputs:

    dumping 0:
        get: function (e,n){var i=t._getOrCreateServiceInstance(e);if(!i&&n!==u.optional)throw new Error("[invokeFunction] unkown service '"+e+"'");return i}
    dumping 1:

Alternative: 
1. Add a key-up handler for the keybinding
Observe: it works for a regular keystroke or chord, but not for the context-menu keyboard key, and the `onContextMenu` event also does not fire.  For example:

```
    editor.onKeyUp(function(e) {
        showEvent('keyup bind - ' + JSON.stringify(arguments));
        console.log('onKeyUp');
        for (let i = 0; i < arguments.length; i++) {
            console.log(`dumping ${i}:`);
            for (let key in arguments[i]) {
                console.log(`    ${key}: ${arguments[i][key]}`);
            }
        }
    });
```

The output is:

    onKeyUp
    dumping 0:
        browserEvent: [object KeyboardEvent]
        target: [object HTMLTextAreaElement]
        ctrlKey: true
        shiftKey: false
        altKey: false
        metaKey: false
        keyCode: 5
        _asKeybinding: [object Object]
        preventDefault: function (){this.browserEvent&&this.browserEvent.preventDefault&&this.browserEvent.preventDefault()}
        stopPropagation: function (){this.browserEvent&&this.browserEvent.stopPropagation&&this.browserEvent.stopPropagation()}
        toKeybinding: function (){return this._asKeybinding}
        equals: function (e){return this._asKeybinding.value===e}
        _computeKeybinding: function (){var e=0;5!==this.keyCode&&4!==this.keyCode&&6!==this.keyCode&&57!==this.keyCode&&(e=this.keyCode);var t=0;return this.ctrlKey&&(t|=l),this.altKey&&(t|=c),this.shiftKey&&(t|=d),this.metaKey&&(t|=h),t|=e,new n.Keybinding(t)}

In summary: There isn't an effective way to use the set of keyboard events to detect the context menu key and comprehend the location of the cursor given the event.

**Comment by alexandrudima** [comment]
(from browsing the source code): The `editor.onContextMenu` is emitted when:
  * we receive a [`'contextmenu'`](https://developer.mozilla.org/en-US/docs/Web/Events/contextmenu) event from the browser
  * when using a touch-enabled device and pressing and holding a touch point for 700ms
In both cases, an `IEditorMouseEvent` is passed in, as this event is generated either via a mouse or via a touch point. This is similar to a native `contextmenu` event, as in the w3c spec.

---

A keyboard listener (e.g. `editor.onKeyUp`) does not get passed in a mouse event, it gets passed in a keyboard event. This is also similar to a native `keyup` event, as in the w3c spec.

---

You can look at how we implemented our context menu (the API has a slightly different shape, but it should feel familiar): https://github.com/Microsoft/vscode/blob/688df5e5cc152e2f5e60d9a3e9fa1ea123dbbbb9/src/vs/editor/contrib/contextmenu/browser/contextmenu.ts
